### PR TITLE
Make LiveIcon subcomponent of Tag

### PR DESCRIPTION
**Pull request by Zystix**
## Why

https://cultureamp.atlassian.net/browse/KZN-2679

## What
Created a subcomponents folder in Tag and move LiveIcon into it, fixed imports.

### CODE REVIEW

File: packages/components/src/Tag/subcomponents/LiveIcon/LiveIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        stroke="currentColor"
+        d="M10.8 5.2c1.533 1.533 1.533 4.067 0 5.667M12.733 3.267c2.6 2.6 2.6 6.8 0 9.4"
+        strokeWidth="1.5"
+        strokeLinejoin="round"
+        strokeLinecap="round"
+      />
+    </>
+  )
+  return <SVG {...props}>{svgContent}</SVG>
+}
```
Comment (HeartSquared) [inline packages/components/src/Tag/subcomponents/LiveIcon/LiveIcon.tsx:67]: Given it's a subcomponent that we will not be exposing, we can just be very deliberate with our composition :)

```suggestion
const LiveIconSvg = (props: IconProps): JSX.Element => (
  <SVG 
    // add the attributes that do not differ between your usages eg. role
    {...props}
  >
    <path
      stroke="currentColor"
      d="M3.266 12.733c-2.6-2.6-2.6-6.866 0-9.466M5.2 10.8c-1.534-1.533-1.534-4.067 0-5.667"
      strokeWidth="1.5"
      strokeLinejoin="round"
      strokeLinecap="round"
    />
    <path fill="currentColor" d="M8 10a2 2 0 1 0 0-4 2 2 0 0 0 0 4" />
    <path
      stroke="currentColor"
      d="M10.8 5.2c1.533 1.533 1.533 4.067 0 5.667M12.733 3.267c2.6 2.6 2.6 6.8 0 9.4"
      strokeWidth="1.5"
      strokeLinejoin="round"
      strokeLinecap="round"
    />
  </SVG>
)
```

Comment (HeartSquared) [reply packages/components/src/Tag/subcomponents/LiveIcon/LiveIcon.tsx:67]: Nit: I'd also suggest that given the reading flow, it is easier when the subcomponent is put above (if not in another file) the consuming component/function.

File: packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx [hunk 2]
```diff
@@ -24,105 +23,16 @@ const meta = {
...
-      classNameOverride={styles.liveIcon_3}
-      width="16"
-      height="16"
-      viewBox="0 0 16 16"
-      fill="none"
-    />
-  </span>
-)
-
 export const LiveIconComponentStory: StoryObj = {
```
Comment (HeartSquared) [inline packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx:26]: Boyscouting if you're up for it:

```suggestion
type Story = StoryObj<typeof meta>

// Update subsequent stories too
export const LiveIconComponentStory: Story = {
```

This ensures type safety where the consuming stories can inherit the props from the meta :)

File: packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx [hunk 3]
```diff
@@ -24,105 +23,16 @@ const meta = {
...
+  render: () => <LiveIcon />,
 }
 
 export const StatusMigration: StoryObj = {
   render: () => (
     <>
       <Tag classNameOverride="gap-4" color="green">
         <span>Tag</span>
-        <LiveIconComponent />
+        <LiveIcon />
```
Comment (HeartSquared) [inline packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx:35]: I haven't reviewed the docs yet, but do let me know when you undraft this PR so I can look at the branch preview.

File: packages/components/src/Tag/subcomponents/LiveIcon/LiveIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+// Changes to this file will be overwritten
+
+import React from 'react'
+import { SVG } from '~components/Icon/subcomponents/SVG'
+import type { IconProps } from '~components/Icon/types'
+import styles from './LiveIcon.module.css'
+
+const LiveIconSvg = (props: IconProps): JSX.Element => (
+  <SVG
+    // add the attributes that do not differ between your usages eg. role
```
Comment (HeartSquared) [inline packages/components/src/Tag/subcomponents/LiveIcon/LiveIcon.tsx:11]: 😂 Maybe I should've put a `@todo` on this lol sorry!

File: packages/components/src/Tag/subcomponents/LiveIcon/LiveIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      stroke="currentColor"
+      d="M10.8 5.2c1.533 1.533 1.533 4.067 0 5.667M12.733 3.267c2.6 2.6 2.6 6.8 0 9.4"
+      strokeWidth="1.5"
+      strokeLinejoin="round"
+      strokeLinecap="round"
+    />
+  </SVG>
+)
+
+export const LiveIcon = (): JSX.Element => (
```
Comment (HeartSquared) [inline packages/components/src/Tag/subcomponents/LiveIcon/LiveIcon.tsx:32]: I'm surprised linting hasn't called you out for it 🤔 Add a displayName to the components - these assist with debugging and Storybook snippets

File: packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx [hunk 6]
```diff
@@ -23,106 +22,18 @@ const meta = {
...
-      classNameOverride={styles.liveIcon_3}
-      width="16"
-      height="16"
-      viewBox="0 0 16 16"
-      fill="none"
-    />
-  </span>
-)
-
-export const LiveIconComponentStory: StoryObj = {
```
Comment (HeartSquared) [inline packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx:65]: Looking at the migration guide, I think you'll want to reinstate this story for the code snippet.

A follow up task would be to speak with our designer Jess to check if we'd actually want people moving forward to just be using the static Future Icon `<Icon name="sensors" isPresentational />` instead, then updating the migration guide again with the decision.

File: packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx [hunk 7]
```diff
@@ -23,106 +22,18 @@ const meta = {
...
 }
 
-export const StatusMigration: StoryObj = {
+export const StatusMigration: Story = {
   render: () => (
     <>
       <Tag classNameOverride="gap-4" color="green">
         <span>Tag</span>
-        <LiveIconComponent />
+        <LiveIcon />
```
Comment (HeartSquared) [inline packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx:36]: Something has gone wrong with the component name somehow 🤔 

image

Comment (Zystix) [reply packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx:36]: Interesting, Winter specifically checked this when we were looking at it together - I'll see whats changed between then and now.

File: packages/components/src/Tag/subcomponents/LiveIcon/LiveIcon.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
+// This file is autogenerated by wrapSVGs.ts
+// Changes to this file will be overwritten
+
+import React from 'react'
+import { SVG } from '~components/Icon/subcomponents/SVG'
+import styles from './LiveIcon.module.css'
+
+type LiveIconSvgProps = { classNameOverride: string }
+
+const LiveIconSvg = ({ classNameOverride }: LiveIconSvgProps): JSX.Element => (
```
Comment (HeartSquared) [inline packages/components/src/Tag/subcomponents/LiveIcon/LiveIcon.tsx:10]: Display name for this too :)

File: packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx [hunk 9]
```diff
@@ -23,106 +22,31 @@ const meta = {
...
-        role="presentation"
-        classNameOverride={styles.liveIcon_3}
-        width="16"
-        height="16"
-        viewBox="0 0 16 16"
-        fill="none"
-      />
-    </span>
-  )
+          import { LiveIcon } from '@kaizen/components/Tag/subcomponents'
```
Comment (HeartSquared) [inline packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx:34]: Haha, not quite! We don't actually want consumers to be using this, so I'd reinstate the original code snippet where consumers need to create their own custom animated live icon.
Moving forward, either we make a proper animated live icon, or we tell people that animated ones are no longer recommended (TBC with design).

Comment (mcwinter07) [reply packages/components/src/__future__/Tag/Tag/_docs/Tag-migration-guide.stories.tsx]: Oh that's my bad, I thought we were exporting from the old tag as a final part of the migration

**Review by HeartSquared — APPROVED** [review]
Nice!

**Pull request merged**